headlamp-k8s/headlamp
# backend: send working list of contexts

If there is a problem with the kubeconfig in any of the context or cluster, headlamp won't be rendered useless. It would throw human readable message to the backend and show the list of working clusters.

- Fixes: #1628
- The original issue was here: https://github.com/headlamp-k8s/headlamp/issues/1349

### Testing

- In your kubeconfig add some context with partial validity. For reference please check [here](https://github.com/headlamp-k8s/headlamp/blob/e13376a49d4191032926373e11d30a1a6c5ff4f7/backend/pkg/kubeconfig/test_data/kubeconfig_partialcontextvalid)
- Run backend `cd backend && go run ./cmd -dev`
- Run frontend `cd frontend && npm start`
- You should see errors in backend and only valid clusters in frontend.
- You can try with different kubeconfig and check.
